### DISCUSSION TITLE
Common sense checks for micrograd

### DISCUSSION THREAD

**Discussion by Jamiewhbee** [Part-II-Computational-Physics/.github#73] [Q&A] [answered]
Some of the results I'm getting don't seem to make sense, so I wanted to check whether my expectations are wrong or whether my code is wrong on the following things.
1. I thought that the decision boundary should separate the inner and outer circles, but I get a wedge shape.
![image](https://user-images.githubusercontent.com/122116209/224382407-935db4f3-da89-45b4-bd35-d8d7ff059bb6.png)

2. Both the gradient descent and Adam optimisers don't have any effect on the accuracy (something that's True in one step will stay True for all steps, and False will stay False).

3. What is the accuracy graph supposed to look like? The plot I'm getting is unintelligible. Maybe I could convert each accuracy from a Boolean array to something else (say, by counting the number of Trues)?
![image](https://user-images.githubusercontent.com/122116209/224383140-79de0b32-85b8-4dc9-b049-7da12b6d5596.png)

**Comment by Jamiewhbee** [comment, marked as answer]
For 1. and 2. I think something went wrong with how I was calculating the SVM loss - I've replaced that section of code with something from the micrograd demo and now it seems to work as expected.
For 3. I'm going to plot the proportion of Trues in each array.

**Comment by Takodachi6969** [comment]
![Screenshot 2023-03-10 191656](https://user-images.githubusercontent.com/102265625/224406225-2c36b3b7-3380-4506-a04c-6bfda6adba7c.png)
mine looks like this if you are interested. 
I have had the same problem with u on the decision part and it was just the micrograd's instruction was bad, you can look into the code and see how the function works, I think what could have happened is that you used the Value function twice and in the end you are only training one parameter out of all 300 of them..
